### DISCUSSION TITLE
Inthe.AM will be shutting down on January 1st, 2024

### DISCUSSION THREAD

**Discussion by coddingtonbear** [coddingtonbear/inthe.am#427] [Announcements]
Ten years ago, I started this project on a whim because I had a simple desire of being able to add tasks to my task list from my phone while I was away from my computer.  At first, functionality was just a basic web interface that let you type out task information into a form, but over the years this grew to include things like support for managing your task list over SMS or even via Trello.  It has come a long way from those early days, and by now Inthe.AM is one of the most popular tools in the Taskwarrior community -- over 7000 folks have created Inthe.AM accounts over that time.

Unfortunately, at this point in my life ten years later, I really don't have enough time to keep up with normal everyday maintenance, let alone investigating possible bugs that folks report.  I've been wrestling with how to handle this going forward over the last year or so, but unfortunately all of the obvious solutions will cost more time than I can give, **so I've made the difficult choice to shut down Inthe.AM at the end of the year**.

Lots of you have over the years recognized that this takes a lot of effort and have offered to donate to help pay for service costs, but what might not have been super obvious is that the user base for this project was always rather small -- never more than 150 active users at a time.  With such a small user base, and the possible expectations of immediate support that accepting money might entail, the project could never reasonably make accepting donations worth the while, even if that offer comes from a generous place.  I do appreciate the offers, though -- thank you again for your support!

Don't worry, though, Taskwarrior tasks are always on your local device so this shutdown will not cause you to lose any data, but you may want to seek out alternative services to use for synchronizing your task lists among multiple devices.  If you happen to run such a service, please mention it in the discussion below (or start a new one) so folks who need that kind of service can know who to turn to.

Cheers, and thank you!

**Comment by jtmart** [comment]
I understand your decision, but this is sad as I am not aware of any alternative hosted service for Taskwarrior still running. Also, have you already deactivated sync for some users? I face this error message since this morning on two of my three devices:

```
Could not connect to inthe.am 53589
Sync failed.  Could not connect to the Taskserver.
```

Best,

Jean-Thomas

**Reply by jtmart** [reply]
Sorry, I found out what the issue was, the error was due to my proxy/firewall settings. Still sad to hear you are shutting this down though: hope someone can take over this amazing project. Taskforce without an easy-to-set-up hosted sync deamon is crippled...

**Comment by ikenichiro** [comment]
Thank you very much for all the years of support and hosting.

I was wondering if the shutdown includes the shutdown of the software development itself or not, I mean will the development of the software (self-hosting opportunities?) continue or not.
Although I do not have much developing knowledge, I'm thinking to try self-hosting for myself since I want to keep on using the nice user interface you developed.

**Reply by coddingtonbear** [reply]
Finding the time to continue development is indeed one of the reasons I'm setting this project down.  The repository will remain up, though, in case others want to continue maintaining it and extending the existing functionality to make self-hosting easier.

As far as how to self-host right now, there is a development server that is fairly easily stood up; see https://intheam.readthedocs.io/en/latest/development/development_environment_setup.html.  For an overview of the structure of Inthe.AM and its component services, see: https://intheam.readthedocs.io/en/latest/development/architecture.html.

**Comment by JonBoyleCoding** [comment]
Thank you very much - this helped me get through the end of my PhD with flying colours. I will migrate to running my own taskd service now that I know a bit more about hosting for my personal use.

It is a really nice user interface - perhaps I'll look into self-hosting it. Are you all right with keeping the repository up post shut-down so people can host their own versions and submit PRs whenever the needs arise?

**Reply by coddingtonbear** [reply]
Yes, the respository will remain up.

**Comment by a-t-0** [comment]
Thank you for the effort you have provided in supporting the usage and adoption of the Taskwarrior ecosystem! I am not associated with Taskwarrior, though enjoy the software thoroughly. 

I am working on a self-hosted alternative over tor [here](https://github.com/HiveMinds/SSL4Tor) and to make it install/setup in a single command [here](https://github.com/HiveMinds/tw-install). I am currently in the process of setting up self-hosted CI's on these repositories to improve the quality and reproducibility, so in the short term it will not be able to provide anywhere near the level of service/quality that I believe in.the.am has shown, however I thought I might point it out as a starting point for people looking (to contribute to)/for an alternative.

**Comment by scheijan** [comment]
I found Inthe.AM soon after I fell in love with Taskwarrior. And @coddingtonbear was super friendly and helpful to get me started.

At the time I also wanted to manage my tasks from within Slack, so I wrote the now long defunct [Slackwarrior](http://slackwarrior.org) wich used Inthe.AM's API to connect my Taskwarrior tasks to a "bot" in Slack.

I have since moved on from Taskwarrior and from Slack, it feels like all this was in a different life. But I have fond memories of this time and I would like to thank you @coddingtonbear for everything you did for the Taskwarrior community. 

Best of luck for you!

**Reply by JonBoyleCoding** [reply]
Just out of interest - what did you move onto?

**Comment by linhns** [comment]
Thanks for all your contributions over the years @coddingtonbear. While it's definitely sucks to hear that you'll be shutting Inthe.AM down, it's a sensible decision.

Good luck in your future endeavors!

**Comment by kmicb** [comment]
SO sad by this but understandable. 😔

**Comment by jmriego** [comment]
Thanks a lot for the years you have supported this! Not easy to be on top of something like this and help users in your own time

**Comment by red-swan** [comment]
Not adding to the discussion here. But thank you. I loved your service. Good luck in your next adventures.

**Comment by coddingtonbear** [comment]
I just a few minutes ago pressed the service's "Power Off" button.  Thank you all again for your support over the years!

**Reply by folofjc** [reply]
Sorry I was late to the party, thanks for everything! Have been using it reliably for years!

**Reply by djmitche** [reply]
I apologize for not saying anything earlier -- I just learned about this! I ran freecinc for a little while, so I definitely empathize!
